core/os/memory.cpp

```diff
@@ -43,41 +44,36 @@ void *operator new(size_t p_size, void *(*p_allocfunc)(size_t p_size)) {
 	return p_allocfunc(p_size);
 }
 
-#include <stdio.h>
-
 #ifdef DEBUG_ENABLED
 size_t Memory::mem_usage = 0;
 size_t Memory::max_usage = 0;
+#define PREPAD true
+#else
+#define PREPAD p_pad_align
```

Comment:
Your last comments gets it right. The preprocessor just replace the defines so the compiler will actually see one thing or the other acording to the build type.

Now, to defend my commit :), the previous code forced prepad in debug builds and anyway you had to review it carefully to understand it.

I wouldn't write such a "robotic" code in other areas, but memory management is low level stuff and anyone wanting to touch this would have anyway to take some time to wrap his head around it (as I had to do).

As an advantage, the reviewed code block shows pretty clearly that things work different across build types and gives some clues about how to interpret the functions below.